subsys/bluetooth/controller/ll_sw/ull_adv_aux.c

```diff
@@ -485,21 +489,25 @@ uint8_t ull_adv_aux_hdr_set_clear(struct ll_adv_set *adv,
 	*(uint8_t *)sec_hdr = 0U;
 
 	/* AdvA flag */
-	/* NOTE: as we will use auxiliary packet, we remove AdvA in
-	 * primary channel. i.e. Do nothing to not add AdvA in the primary
-	 * PDU. Connectable or scannable advertising always has AdvA in aux.
+	/* NOTE: as we will use auxiliary packet, we remove AdvA in primary
+	 * channel, i.e. do nothing to not add AdvA in the primary PDU.
+	 * AdvA can be either set explicitly (i.e. needs own_addr_type to be
+	 * set), can be copied from primary PDU (i.e. adding AD to existing set)
+	 * or can be copied from previous secondary PDU.
 	 */
-	if (pri_hdr_prev.adv_addr) {
-		pri_dptr_prev += BDADDR_SIZE;
-
-		/* Prepare to add AdvA in secondary PDU */
-		sec_hdr->adv_addr = 1;
+	sec_hdr->adv_addr = 1;
+	if (sec_hdr_add_fields & ULL_ADV_PDU_HDR_FIELD_ADVA) {
+		uint8_t own_addr_type = *(uint8_t *)value;
 
-		/* NOTE: AdvA is filled at enable */
+		sec_pdu->tx_addr = own_addr_type & 0x1;
+	} else if (pri_hdr_prev.adv_addr) {
+		pri_dptr_prev += BDADDR_SIZE;
```

Comment:
Should `pri_dptr_prev`  be incremented when setting addr type explicitly? Are there possibilities that there was an address in primary PDU, and explicit addr type would be set?

Reply:
ah right. we can have AdvA in prev primary and set own addr explicitly if we set parameters to change set from nconn-nscan to e.g. nconn-scan. will fix.